aioaria2/client.py

```
@@ -878,36 +895,42 @@ async def new(
-    async def send_request(self, req_obj: Dict[str, Any]) -> Union[Dict[str, Any], str, NoReturn]:  # type: ignore
-        try:
-            await self.client_session.send_json(req_obj, dumps=self.dumps)
-            data = await ResultStore.fetch(
-                req_obj["id"], self.kw.get("timeout", None) or 10.0
-            )
-            return data["result"]
-        except KeyError:  # 'error':xxx
-            raise Aria2rpcException(f"unexpected result: {data}")
-        except Aria2rpcException as err:
-            if not self.closed and "timeout" in err.msg:
+    async def send_request(self, req_obj: dict[str, Any]) -> dict[str, Any] | str | NoReturn:  # type: ignore
```

Ruff (0.16.1)

[warning] `NoReturn | T` is equivalent to `T`

Remove `NoReturn`

(RUF020)